rules/prefer-string-slice.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+const eslintTemplateVisitor = require('eslint-template-visitor');
+const getDocsUrl = require('./utils/get-docs-url');
+
+const templates = eslintTemplateVisitor();
+
+const objectVariable = templates.variable();
+
+const substrCallTemplate = templates.template`${objectVariable}.substr()`;
```

Comment:
Would be cool to be able to specify the method dynamically too and then have the be able to automatically generate the fixer by saying what we want to replace.

```suggestion
const substrCallTemplate = templates.template`${objectVariable}.${method}()`;
```

I guess it would be much more complicated if the arguments needs to change, but in this case it shouldn't be too hard.

I like how easy [jscodeshift](https://github.com/facebook/jscodeshift) makes it to create Codemods: https://github.com/avajs/ava-codemods/blob/1b3f183d58834e5b9f24340ba8563efa5378d30e/lib/error-to-iferror.js#L7 You might find some inspiration in it.

Reply:
You might also find https://github.com/jfmengels/enhance-visitors useful. We use it heavily in `eslint-plugin-ava`.

Reply:
Yes, this would be nice to have, but this does not fit well withing `eslint-template-visitor`. Maybe I could provide this with another package with an API like this:

```js
fix: fixer => jscodeshiftFixer(fixer, node).findMethodCalls('substr').renameTo('slice')
```